### PR TITLE
Constellation create with Terraform (GCP)

### PR DESCRIPTION
**Pull request by katexochen**
### Proposed change(s)

- `constellation create` now used Terraform internally to create and destroy the needed resources
- Metadata like uid and role are now taken from the metadata API instead of parsing them from the instance name
- Serial port is enabled in debug mode
- Remove public IPs for instances
- Multi-zone instance groups for workers and control planes

### Checklist

- [x] Update [CHANGELOG.md](https://github.com/edgelesssys/constellation/blob/main/CHANGELOG.md)
- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Add documentation to the Terraform module, especially regarding the required values, for people that want to write their own
- [x] Link to Milestone
- [ ] Build new operator on main

### Future ToDos (not this PR)

- [ ]  Write `.tfvars` file with [hashicorp/hcl ](https://github.com/hashicorp/hcl)
- [ ] Maybe generate Go structs for variables from `variables.tf` file
- [ ] Use labels instead of metadata on instances, refactor metadata code

### CODE REVIEW

**Review by Nirusu — COMMENTED** [review]
I'll test this again more in-depth when it's ready for review. Some nits.

**Comment by katexochen** [comment]
Image to test: `projects/constellation-331613/global/images/debug-tf-gcp-paul`

File: cli/internal/cloudcmd/create.go [hunk 1]
```diff
@@ -25,28 +23,21 @@ import (
...
 }
 
 // NewCreator creates a new creator.
 func NewCreator(out io.Writer) *Creator {
 	return &Creator{
 		out: out,
-		newGCPClient: func(ctx context.Context, project, zone, region, name string) (gcpclient, error) {
-			return gcpcl.NewInitialized(ctx, project, zone, region, name)
+		newTerraformClient: func(ctx context.Context) (terraformClient, error) {
+			return terraform.New(ctx, cloudprovider.GCP)
```
Comment (Nirusu) [inline cli/internal/cloudcmd/create.go:36]: This breaks QEMU and other CSPs. The cloud provider should be passed as a variable to the function.

**Review by Nirusu — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 <!-- For changes in existing functionality.  -->
 - Autoscaling is now directly managed inside Kubernetes, by the Constellation node operator.
+- The `constellation create` on GCP now uses Terraform to create and destroy cloud resources.
```
Comment (Nirusu) [inline CHANGELOG.md:32]: You can also add the GCP no public IP change here (and that SSH now needs to be manually allowed before it can be used).

**Review by daniel-weisse — APPROVED** [review]
Works on my cloud subscription :+1:

**Review by Nirusu — APPROVED** [review]
Approved under duress ;) 
Just kidding, looks good to me.

**Pull request merged**